lib/Cleanup2.ml

```diff
@@ -95,11 +95,10 @@ let remove_implicit_array_copies =
                         (self#remove_assign n (lift lifting_index lhs_i) (lift lifting_index e)
                            (nest lifting_index (array_index + 1) es))
                   | _ ->
-                      with_type e2.typ
-                        (ELet
-                           ( H.sequence_binding (),
-                             H.with_unit (EAssign (lift lifting_index lhs_i, lift lifting_index e)),
-                             nest (lifting_index + 1) (array_index + 1) es )))
+                      with_type e2.typ (self#visit_ELet ((),TUnit)
```

Comment:
why do we need to go through visit_ELet below again?

Reply:
the intent of this code (as I recall it) is that remove_assign is not supposed to recurse back into the visitor -- it's for base cases of desugaring assignments